sources/logdog/src/log_request.rs

```diff
@@ -91,19 +94,31 @@ where
     P: AsRef<Path>,
 {
     let request = request.as_ref();
-    // get the first and second token: i.e. mode and output filename, put the remainder of the
-    // log request into the instructions field (or default to an empty string).
-    let mut iter = request.splitn(3, ' ');
-    let req = LogRequest {
-        mode: iter.next().context(error::ModeMissing)?,
-        filename: iter.next().context(error::FilenameMissing { request })?,
-        instructions: iter.next().unwrap_or(""),
+    let mut conf = request.splitn(2, ' ');
+    let mode = conf.next().context(error::ModeMissing)?;
+    let req = if mode == "glob" {
+        // for glob request format is "glob <pattern>"
+        LogRequest {
+            mode,
+            filename: "",
+            instructions: conf.next().context(error::PatternMissing)?,
+        }
+    } else {
+        // get the first and second token: i.e. mode and output filename, put the remainder of the
+        // log request into the instructions field (or default to an empty string).
+        let mut iter = request.splitn(3, ' ');
+        LogRequest {
+            mode: iter.next().context(error::ModeMissing)?,
```

Comment:
```suggestion
        // Get the second token (output filename) and put the remainder of the
        // log request into the instructions field (or default to an empty string).
        let mut iter = conf.splitn(2, ' ');
        LogRequest {
            mode,
```
Since we already have that information...